gemini/sample-apps/llamaindex-rag/src/app/main.py

```diff
@@ -0,0 +1,484 @@
+from datetime import datetime
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Any, Iterator, List, Optional, Sequence, Tuple, Union, cast
+import uuid
+
+from common.common import DATA_PATH
+from common.utils import download_blob
+from datasets import Dataset
+from fastapi import Depends, FastAPI, HTTPException
+from google.api_core.client_options import ClientOptions
+from google.api_core.exceptions import PermissionDenied
+import google.auth
+import google.auth.transport.requests
+from google.cloud import (
+    aiplatform,
+    firestore,
+    firestore_admin_v1,
+    secretmanager,
+    storage,
+)
+from google.oauth2 import service_account
+from langchain_google_vertexai import ChatVertexAI, VertexAIEmbeddings
+import pandas as pd
+from pydantic import BaseModel
+
+# Ragas must be 0.9.1
+from ragas import evaluate
+from ragas.metrics import (
+    answer_correctness,
+    answer_relevancy,
+    answer_similarity,
+    context_precision,
+    context_recall,
+    context_relevancy,
+    faithfulness,
+)
+from src.rag.evaluate import LLMEvaluator, write_results_to_bq
+import uvicorn
+import yaml
+
+logging.basicConfig(
+    filename="eval.log", encoding="utf-8", level=logging.INFO
+)  # Set the desired logging level
+logger = logging.getLogger(__name__)
+
+config_path = os.path.join(
+    os.path.dirname(__file__), "..", "..", "common", "config.yaml"
+)
+with open(config_path, "r") as config_file:
+    config = yaml.safe_load(config_file)
```

Comment:
Don't you already have a method that does this?

Reply:
fixed by - [commit](https://github.com/GoogleCloudPlatform/generative-ai/commit/2a1a2a975748aee73c720042b53f6c0fe0580134)